Packs/AzureKeyVault/Integrations/AzureKeyVault/AzureKeyVault.py

```diff
@@ -0,0 +1,1304 @@
+
+        if network_acls:
+            properties["networkAcls"] = network_acls
+
+        return properties
+
+    def get_entities_independent_of_pages(self, first_page: Dict[str, Any], limit: int, offset: int,
+                                          resource: str = MANAGEMENT_RESOURCE) -> List[dict]:
+        """
+        List the entities independent of azure's page size.
+
+        Args:
+            first_page (Dict[str, Any]): The first list of entities which returned by the first API call.
+            limit (int): limit on the number of entities to retrieve to the user.
+            offset (int): first index to return from.
+            resource (str): Azure resource. Default is MANAGEMENT_RESOURCE.
+
+        Returns:
+            List[Dict,Any]: List of Key Vaults/Keys/Secrets/Certificates.
+        """
+        entities = first_page.get('value')
+        next_page_url = first_page.get('nextLink')
+        # more entities to get
+        while next_page_url and len(entities) < offset + limit:
+            next_page_url = next_page_url.replace('api-version', 'api')
+            response = self.http_request(
+                'GET', full_url=next_page_url, resource=resource)
+
+            entities = entities + response.get('value', [])
+            next_page_url = response.get('nextLink')
+        if offset > len(entities):
+            return []
+        return entities[offset:limit + offset]
+
+    def get_secret_credentials(self, key_vault_name: str, secret_name: str):
+        try:
+            response = self.get_secret_request(key_vault_name, secret_name, '')
+            secret_value = response['value']
+            return {
+                "user": secret_name,
+                "password": secret_value,
+                "name": f'{key_vault_name}/{secret_name}'
+            }
+        except Exception:
+            return None
+
+
+''' INTEGRATIONS COMMANDS'''
+
+
+def create_or_update_key_vault_command(client: KeyVaultClient, args: Dict[str, Any]) -> CommandResults:
+    """
+    Create or update key vault command.
+
+    Args:
+        client (KeyVaultClient):Azure Key Vault API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    vault_name = args.get('vault_name')
+    location = args.get('location', 'westus')
+    sku_name = args.get('sku_name', 'standard')
+    object_id = args.get('object_id')
+
+    # access policy arguments
+    keys = argToList(args.get('keys', ['get', 'list', 'update', 'create', 'import', 'delete', 'recover', 'backup',
+                                       'restore']))
+
+    secrets = argToList(args.get('secrets', ['get', 'list', 'set', 'delete', 'recover', 'backup', 'restore']))
+    certificates = argToList(
+        args.get('certificates', ['get', 'list', 'update', 'create', 'import', 'delete', 'recover', 'backup', 'restore',
```

Comment:
Each variable (keys/secrets/certificates) represents permissions list.
I think it will be better to change the variables names for clearer names. For example:
keys -> keys_permissions.